tribe-ext-extension-template.php

```diff
@@ -204,6 +222,47 @@ private function php_version_check() {
 			return true;
 		}
 
+		/**
+		 * Check if we have the needed view. Admin notice if we don't and user should see it.
+		 *
+		 * @return bool
+		 *
+		 * @todo Remove method if extension works with both V1 and V2 views
+		 */
+		private function v2_views_check() {
+			$view_required_version = $this->view_needed;
+
+			$view_breaks_version = $view_required_version == 'V1 legacy' ? 'V2 updated' : 'V1 legacy';
+
+			$show_warning = (
+				$view_required_version == 'V1 legacy' && tribe_events_views_v2_is_enabled()
+				|| $view_required_version == 'V2 updated' && ! tribe_events_views_v2_is_enabled()
+			)
+				? true
+				: false;
+
+			if ( function_exists( 'tribe_events_views_v2_is_enabled' ) ) {
+				if ( is_admin() && current_user_can( 'activate_plugins' ) && $show_warning ) {
+					$message = '<p>';
+					$message .= sprintf(
+						__(
+							'%s requires %s views to work. It will not work with %s views.', PLUGIN_TEXT_DOMAIN
+						),
+						$this->get_name(),
+						$view_required_version,
+						$view_breaks_version,
+					);
+					$message .= '</p>';
+
+					tribe_notice( PLUGIN_TEXT_DOMAIN . '-views-version', $message, [ 'type' => 'warning' ] );
+				}
+
+				return false;
+			}
```

Comment:
To avoid large indentation.

```suggestion
			if ( ! function_exists( 'tribe_events_views_v2_is_enabled' ) ) {
				return true;
			}
			
			if ( is_admin() && current_user_can( 'activate_plugins' ) && $show_warning ) {
					$message = '<p>';
					$message .= sprintf(
						__(
							'%s requires %s views to work. It will not work with %s views.', PLUGIN_TEXT_DOMAIN
						),
						$this->get_name(),
						$view_required_version,
						$view_breaks_version,
					);
					$message .= '</p>';

					tribe_notice( PLUGIN_TEXT_DOMAIN . '-views-version', $message, [ 'type' => 'warning' ] );
				}
			return false;
```